Adaptors/Amqp/src/SessionAmqp.cs

```diff
@@ -0,0 +1,133 @@
+
+    public ILogger Logger { get; set; }
+
+    public SessionAmqp(Options.Amqp options,
+                       ILogger      logger)
+    {
+      Options = options;
+      Logger  = logger;
+
+    }
+
+    private int retriesReconnect_ = 5;
+
+    private static void OnCloseConnection(IAmqpObject sender,
+                                          Error       error,
+                                          ILogger     logger)
+    {
+      if (error == null)
+        logger.LogInformation("AMQP Connection closed with no error");
+      else
+      {
+        logger.LogWarning("AMQP Connection closed with error: {0}",
+                        error.ToString());
+      }
+    }
+
+    public async Task<SessionAmqp> OpenConnection()
+    {
+      if (Session != null)
+      {
+        await Session.CloseAsync()
+                     .ConfigureAwait(false);
+        if (retriesReconnect_ <= 0)
+          throw new AmqpException(new Error($"After 5 retries. Fail to reconnect to Amqp"));
+
+        --retriesReconnect_;
+
+      }
+
+      var address = new Address(Options.Host,
+                                Options.Port,
+                                Options.User,
+                                Options.Password,
+                                scheme: Options.Scheme);
+
+      var connectionFactory = new ConnectionFactory();
+      if (Options.Scheme.Equals("AMQPS"))
+      {
+        connectionFactory.SSL.RemoteCertificateValidationCallback = delegate(object          sender,
+                                                                             X509Certificate certificate,
+                                                                             X509Chain       chain,
+                                                                             SslPolicyErrors errors)
+                                                                    {
+                                                                      switch (errors)
+                                                                      {
+                                                                        case SslPolicyErrors.RemoteCertificateNameMismatch when Options.AllowHostMismatch:
+                                                                        case SslPolicyErrors.None:
+                                                                          return true;
+                                                                        default:
+                                                                          Logger.LogError("SSL error : {error}",
+                                                                                          errors);
+                                                                          return false;
+                                                                      }
+                                                                    };
+      }
+
+      var retries = 1;
+      while (true)
```

Comment:
Avoid 'while(true)' 
Use 'for(Var retry=0; retry < maxRetry ; ++retry)' instead